src/dgrid/DgridWrapper.ts

```diff
@@ -0,0 +1,658 @@
+			if (selection) {
+				grid.on('dgrid-select', (event: DgridSelectionEvent) => {
+					const onSelect = properties.onSelect;
+					onSelect && onSelect(buildSelectEvent(event), grid.selection);
+				});
+				grid.on('dgrid-deselect', (event: DgridSelectionEvent) => {
+					const onDeselect = properties.onDeselect;
+					onDeselect && onDeselect(buildSelectEvent(event), grid.selection);
+				});
+			}
+			if (columnHider) {
+				grid.on('dgrid-columnstatechange', (event: DgridColumnHiderEvent) => {
+					const onColumnStateChange = properties.onColumnStateChange;
+					onColumnStateChange && onColumnStateChange(buildColumnStateChange(event));
+				});
+			}
+			if (columnReorder) {
+				grid.on('dgrid-columnreorder', (event: DgridColumnReorderEvent) => {
+					const onColumnReorder = properties.onColumnReorder;
+					onColumnReorder && onColumnReorder(buildColumnReorder(event));
+				});
+			}
+			if (columnResizer) {
+				grid.on('dgrid-columnresize', (event: DgridColumnResizeEvent) => {
+					const onColumnResize = properties.onColumnResize;
+					onColumnResize && onColumnResize(buildColumnResize(event));
+				});
+			}
+			if (keyboard) {
+				grid.on('dgrid-cellfocusin', (event: DgridCellFocusEvent) => {
+					const onCellFocusIn = properties.onCellFocusIn;
+					onCellFocusIn && onCellFocusIn(buildCellFocus(event));
+				});
+
+				grid.on('dgrid-cellfocusout', (event: DgridCellFocusEvent) => {
+					const onCellFocusOut = properties.onCellFocusOut;
+					onCellFocusOut && onCellFocusOut(buildCellFocus(event));
+				});
+			}
+		}
+		const { onSort, onError } = properties;
+
+		if (onSort) {
+			grid.on('dgrid-sort', (event: DgridSortEvent) => {
+				onSort(buildSort(event));
+			});
+		}
+
+		if (onError) {
+			grid.on('dgrid-error', (event: DgridErrorEvent) => {
+				onError(buildError(event));
+			});
+		}
+	}
+}
+
+const keyPrefix = 'dgridWrapper';
+
+@diffProperty('columns', columnsDiff)
+export class DgridWrapper extends WidgetBase<DgridWrapperProperties> {
+	private key = 0;
+	private _gridState: DgridState;
+
+	protected render(): DNode | DNode[] {
+		const changedPropertyKeys = this.changedPropertyKeys;
+		// Some properties require the dgrid grid to be destroyed and recreated because a different
+		// combination of mixins is required.  Some properties when changed can be handled by an existing
+		// grid.  Determine if a new grid is needed or not.  If a new grid is needed, generate a new
+		// key and let the vdom throw away the old grid.
+		if (changedPropertyKeys.indexOf('features') >= 0) {
+			this.key++;
+		} else {
+			const gridConstructionKeys = changedPropertyKeys.filter((key) => {
+				return constructionKeys.indexOf(key) >= 0;
+			});
+			if (gridConstructionKeys.length) {
+				this.key++;
+			}
+		}
+		return w(DgridInnerWrapper, {
+			key: `${keyPrefix}${this.key}`,
+			...this.properties,
+			gridState: this._gridState,
+			onGridState: (gridState: DgridState) => {
+				this._gridState = gridState;
```

Comment:
I don't believe so. This is just supposed to keep track of the latest internal grid state so that if and when the grid needs to be destroyed we'll have that ready to pass back in.